### PR TITLE
[ENG-6162][eas-cli] Change choice formatting when selecting build to submit

### PR DESCRIPTION
**Pull request by szdziedzic**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

https://linear.app/expo/issue/ENG-6162/output-uses-release-channel-when-selecting-a-build-to-submit

# How

Use different formatting for choices suggested by @jonsamp in the [Linear task](https://linear.app/expo/issue/ENG-6162/output-uses-release-channel-when-selecting-a-build-to-submit).

## Before:
image

## After:
Screenshot 2022-11-03 at 12 09 17
Screenshot 2022-11-03 at 12 09 17

# Test Plan

Test manually.

### CODE REVIEW

**Review by szdziedzic — COMMENTED** [review]
@brentvatne Do we want to do this for every prompt in the CLI or just this one?
Screenshot 2022-11-03 at 12 11 39

File: packages/eas-cli/src/submit/ArchiveSource.ts [hunk 1]
```diff
@@ -304,38 +305,31 @@ async function handleBuildListSourceAsync(
...
     field ? chalk.bold(field) : chalk.dim('Unknown');
 
   const buildDate = new Date(updatedAt);
-  const maybeRuntimeVersion = runtimeVersion ? `Runtime: ${formatValue(runtimeVersion)}` : null;
-  const maybeSdkVersion = sdkVersion ? `SDK: ${formatValue(sdkVersion)}` : null;
-  const appBuildVersionString = `${
-    platform === AppPlatform.Android ? 'Version code' : 'Build number'
-  }: ${formatValue(appBuildVersion)}`;
+
+  const formatedCommitData =
```
Comment (dsokal) [inline packages/eas-cli/src/submit/ArchiveSource.ts:321]:
```suggestion
  const formattedCommitData =
```

**Review by dsokal — APPROVED** [review]
Just curious because I don't understand one thing - which line changes the underline behavior?

File: packages/eas-cli/src/submit/ArchiveSource.ts (hunk 1 above)
Comment (szdziedzic) [reply packages/eas-cli/src/submit/ArchiveSource.ts:321]: Just checked the prompts library [documentation](https://www.npmjs.com/package/prompts#selectmessage-choices-initial-hint-warn) and it seems like we can't get rid of the underline effect using its API 🤔. I assumed it is possible, although in reality, it's not.

**Comment by szdziedzic** [comment]
@dsokal Just checked the prompts library [documentation](https://www.npmjs.com/package/prompts#selectmessage-choices-initial-hint-warn) and it seems like we can't get rid of the underline effect using its API 🤔. I assumed it is possible, although, in reality, it was not.

**Comment by dsokal** [comment]
I think you misunderstood me :D 

Screenshot 2022-11-03 at 14 24 44

Just compare the two outputs. In the first screenshot, there is an underlining under whitespace.

**Comment by szdziedzic** [comment]
The same happens when you use spaces.

**Review by wkozyra95 — APPROVED** [review]

**Comment by szdziedzic** [comment]
Do you have any ideas on how we can make the situation with the underline effect better? 🤔

**Review by szdziedzic — COMMENTED** [review]
This is how it looks like in the version with the single-line title and multiline description:

https://user-images.githubusercontent.com/55145344/201054563-52a33b40-35ce-4765-9a6c-27ffa7798f3f.mov

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/submit/ArchiveSource.ts [hunk 2]
```diff
@@ -304,42 +305,36 @@ async function handleBuildListSourceAsync(
...
+  const formattedCommitData =
+    gitCommitHash && gitCommitMessage
+      ? `${chalk.dim(gitCommitHash.slice(0, 7))} "${chalk.bold(gitCommitMessage)}"`
+      : 'Unknown';
+
+  const title = `ID: ${chalk.dim(id)} (${chalk.dim(`${fromNow(buildDate)} ago`)})`;
+
+  const description = [
+    `\tProfile: ${formatValue(buildProfile)}`,
+    `\tChannel: ${formatValue(releaseChannel)}`,
```
Comment (dsokal) [inline packages/eas-cli/src/submit/ArchiveSource.ts:330]: Is it "release channel" (legacy) or "channel" (eas update)? If the former, then maybe we should name this "Release channel"? Also, maybe it's worth printing the "channel" as well?

File: packages/eas-cli/src/submit/ArchiveSource.ts [hunk 3]
```diff
@@ -304,42 +305,36 @@ async function handleBuildListSourceAsync(
...
-  const maybeRuntimeVersion = runtimeVersion ? `Runtime: ${formatValue(runtimeVersion)}` : null;
-  const maybeSdkVersion = sdkVersion ? `SDK: ${formatValue(sdkVersion)}` : null;
-  const appBuildVersionString = `${
-    platform === AppPlatform.Android ? 'Version code' : 'Build number'
-  }: ${formatValue(appBuildVersion)}`;
-
-  const title = [
-    `ID: ${chalk.dim(id)}, Finished at: ${chalk.bold(buildDate.toLocaleString())}`,
-    [
-      `\tApp version: ${formatValue(appVersion)}, ${appBuildVersionString}`,
```
Comment (dsokal) [inline packages/eas-cli/src/submit/ArchiveSource.ts:331]: Why did you remove it?

File: packages/eas-cli/src/submit/ArchiveSource.ts [hunk 4]
```diff
@@ -304,42 +305,36 @@ async function handleBuildListSourceAsync(
...
-      maybeRuntimeVersion,
-      maybeSdkVersion,
-    ]
-      .filter(it => it != null)
-      .join(', '),
-    `\tProfile: ${formatValue(buildProfile)}, Release channel: ${formatValue(releaseChannel)}`,
-    `\tAuthored by: ${formatValue(initiatingActor?.displayName)}`,
```
Comment (dsokal) [inline packages/eas-cli/src/submit/ArchiveSource.ts:338]: Same question

File: packages/eas-cli/src/submit/ArchiveSource.ts (hunk 2 above)
Comment (szdziedzic) [reply packages/eas-cli/src/submit/ArchiveSource.ts:330]: Thanks, I meant to use EAS Update `build.channel` as data for `Channel`. We want to switch to only using the `Channel` vs `Release Channel`.

File: packages/eas-cli/src/submit/ArchiveSource.ts (hunk 3 above)
Comment (szdziedzic) [reply packages/eas-cli/src/submit/ArchiveSource.ts:331]: Jon (who is the issue author), suggested that only these fields should be displayed:

Screenshot 2022-11-10 at 11 26 01

**Review by dsokal — APPROVED** [review]

**Comment by szdziedzic** [comment]
/changelog-entry chore Improve choice formatting when selecting a build to submit in the `eas submit` command
